src/databricks/sql/auth/retry.py

```diff
@@ -0,0 +1,378 @@
+import logging
+import time
+import typing
+from enum import Enum
+from typing import List, Optional, Tuple, Union
+
+from urllib3 import BaseHTTPResponse  # type: ignore
+from urllib3 import Retry
+from urllib3.util.retry import RequestHistory
+
+from databricks.sql.exc import (
+    CursorAlreadyClosedError,
+    MaxRetryDurationError,
+    NonRecoverableNetworkError,
+    SessionAlreadyClosedError,
+    UnsafeToRetryError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CommandType(Enum):
+    EXECUTE_STATEMENT = "ExecuteStatement"
+    CLOSE_SESSION = "CloseSession"
+    CLOSE_OPERATION = "CloseOperation"
+    OTHER = "Other"
+
+    @classmethod
+    def get(cls, value: str):
+        value_name_map = {i.value: i.name for i in cls}
```

Comment:
This is kind of a weird way to use an enum, since you're not actually enumerating.  I'm assuming you have some place where you're taking a string value from external I/O?  If not, worth thinking about how we can use this as a true enum.  If so, we shouldn't need to build the value map on each call, so move this out and up.

Reply:
Yeah this one's weird. But I think how it sits is the right approach. 

The point here is to avoid passing around strings (fragile) and assert types.

`CommandType.get()` is only called by `thrift_backend.py` as a way to normalise the name of the command it is preparing to execute, which it communicates to the retry policy.